query/stdlib/influxdata/influxdb/storage.go

```diff
@@ -149,10 +149,15 @@ type ReadWindowAggregateSpec struct {
 	// TODO(issue #17784): add attributes for the window aggregate spec.
 }
 
+// WindowAggregateCapability describes what is supported by WindowAggregateReader.
+type WindowAggregateCapability struct{}
+
 // WindowAggregateReader implements the WindowAggregate capability.
 type WindowAggregateReader interface {
 	// HasWindowAggregateCapability will test if this Reader source supports the ReadWindowAggregate capability.
-	HasWindowAggregateCapability(ctx context.Context) bool
+	// If WindowAggregateCapability is passed to the method, then the struct
+	// is filled with a detailed list of what the RPC call supports.
+	HasWindowAggregateCapability(ctx context.Context, capability ...*WindowAggregateCapability) bool
```

Comment:
Why `WindowAggregateCapability` is not directly returned along with that boolean. or I am thinking maybe the function can just return a `WindowAggregateCapability` type. a non-nil value means it is supported?

Reply:
There's no reason why it doesn't. We could potentially just do that. The main reason is I wanted to be able to do something like this:

```
var capability WindowAggregateCapability
if !HasWindowAggregateCapability(ctx, &capability) {
    return
}
```

If you return more than one parameter, you have to assign it to a variable to use the boolean result.

Reply:
Ah I see.